src/storage/checkpoint_manager.cpp

```diff
@@ -519,8 +528,22 @@ void CheckpointReader::ReadTableData(ClientContext &context, Deserializer &deser
 	// This is written in "SingleFileTableDataWriter::FinalizeTable"
 	auto table_pointer = deserializer.ReadProperty<MetaBlockPointer>(101, "table_pointer");
 	auto total_rows = deserializer.ReadProperty<idx_t>(102, "total_rows");
+
+	// for current duckdb file versions, this will read a vector with exactly one {-1, 42} index pointer
 	auto index_pointers = deserializer.ReadProperty<vector<BlockPointer>>(103, "index_pointers");
 
+	if (!index_pointers.empty() && !index_pointers.front().IsValid() && index_pointers.front().offset == 42) {
+		deserializer.ReadList(104, "index_storage_infos", [&](Deserializer::List &list, idx_t i) {
```

Comment:
Can we remove the `deserializer.ReadList` here and use the regular deserialization code?